subscriber.go

```diff
@@ -582,6 +582,12 @@ func (h *handler) handleAsync(ctx context.Context, nextCid cid.Cid, ss ipld.Node
 func (h *handler) handle(ctx context.Context, nextCid cid.Cid, sel ipld.Node, wrapSel, updateLatest bool, syncer Syncer) error {
 	if wrapSel {
 		sel = ExploreRecursiveWithStopNode(selector.RecursionLimitNone(), sel, h.latestSync)
+
+		if h.latestSync != nil && h.latestSync.(cidlink.Link).Cid == nextCid {
```

Comment:
Shouldn't this be done whether or not `wrapSel` is true?  If so, then move to top of function, and if not then please comment.